### ISSUE TITLE
Define source ip for check_ping. [sf#3552828]

### ISSUE THREAD

**Issue by monitoring-user** [monitoring-plugins/monitoring-plugins#1136]
Submitted by calestyo on 2012-08-01 00:53:51

Hi.

This is from the old / soon to be disabled again Nagios Plugins bug tracker that used to be at Nagios itself.
I've just copied this bug over. I'm not the original reporter and have no idea about the thoughts about this bug.
## bva:

There are thousand of inet aliases on same interface on Nagios host. And some of switches which I need to check, accepts ICMP only from certain internal addresses, but default route policy forward packets through wrong gateway. I couldn't change routing and settings on switches.

The solution [for me] is to specify the source address for check_ping.

Used nagios-plugins-1.4.15

[nagios]> ./check_ping -H 192.168.17.114 -w 3000.0,80% -c 5000.0,100% -p 1
PING CRITICAL - Packet loss = 100%|rta=5000.000000ms;3000.000000;5000.000000;0.000000 pl=100%;80;100;0
[nagios]>
[nagios]> route get 192.168.17.114
   route to: 192.168.17.114
destination: 192.168.17.112
       mask: 255.255.255.252
    gateway: cat1
  interface: vlan1
      flags: <UP,GATEWAY,DONE,PROTO1>
 recvpipe sendpipe ssthresh rtt,msec rttvar hopcount mtu expire
       0 0 0 0 0 0 1500 0
[nagios]>
[nagios]> ./check_ping2 -S 192.168.150.3 -H 192.168.17.114 -w 3000.0,80% -c 5000.0,100% -p 1
PING OK - Packet loss = 0%, RTA = 3.67 ms|rta=3.668000ms;3000.000000;5000.000000;0.000000 pl=0%;80;100;0

**Comment by riskersen** [comment]
The attached file (corrected URL https://www.monitoring-plugins.org/attachments/450268-nagios-plugins.patch) is lacking the recognition of configure script, so there are more adjustments needed

**Comment by RincewindsHat** [comment]
`check_icmp` has a source address option and I worry about different `ping` implementations not having this options.
Therefore I am closing this issue, but if someone comes up with a reasonable set of changes which address these problems I would be open to the idea.
